### PR TITLE
Issue72 openeo geotrellis extensions add coverage for projection metadata

### PR DESCRIPTION
**Pull request by JohanKJSchreurs**
Adding coverage to check that the STAC projection metadata is correctly added to the job's metadata.

This would catch missing projection metadata, for example the problem that was reported in this comment:
https://github.com/Open-EO/openeo-geotrellis-extensions/issues/72#issuecomment-1491745761

### Related PRs:

When this PR is merged the integration tests will fail because the STAC projection metadata is currently not being added correctly.

The PR that fixes this in openeo-geopyspark-driver is this one: 
https://github.com/Open-EO/openeo-geopyspark-driver/pull/389

To avoid that other, unrelated PRs of the openeo-geopyspark-driver cannot be deployed after they are merged, it would be best to first merge https://github.com/Open-EO/openeo-geopyspark-driver/pull/389 before you merge this PR (https://github.com/Open-EO/openeo-geopyspark-integrationtests/pull/5) .

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -296,6 +297,164 @@ def test_cog_execute_batch(auth_connection, tmp_path):
...
+    # Verify projection metadata.
+    job_metadata = job_results.get_metadata()
+    assert job_metadata == DictSubSet(
+        {
+            "epsg": 4326,
+            "assets": {
+                "openEO_2017-11-21Z_N51E002.tif": DictSubSet(
+                    {
+                        "proj:shape": [365, 728],
+                        "proj:bbox": [1.9995067, 51.0012761, 3.0020091, 52.0010318],
```
[the remark this answers is not in the document]
Comment (JohanKJSchreurs) [reply tests/test_integration.py]: Done.

File: tests/test_integration.py [hunk 2]
```diff
@@ -296,6 +297,164 @@ def test_cog_execute_batch(auth_connection, tmp_path):
...
+
+
+def assert_projection_metadata_present(metadata: dict) -> dict:
+    """Check all elements of the STAC projection metadata.
+
+    If the assert fails then include the report in the assert message so we have
+    some information to start troubleshooting.
+    """
+    report = validate_projection_metadata(metadata)
+    from pprint import pformat
```
[the remark this answers is not in the document]
Comment (JohanKJSchreurs) [reply tests/test_integration.py]: Done.

File: tests/test_integration.py [hunk 3]
```diff
@@ -336,6 +510,19 @@ def elapsed():
     raise RuntimeError("reached max poll time: {e} > {m}".format(e=elapsed(), m=max_poll_time))
 
 
+def assert_job_status(job: BatchJob, expected_status):
+    # If the next assert is going to fail, then first show the logs of the job.
+    if job.status != expected_status:
```
Comment (soxofaan) [inline tests/test_integration.py:515]:
```suggestion
    if job.status() != expected_status:
```

Also further in this function, but I would store it in a local var for simplicity

File: tests/test_integration.py [hunk 4]
```diff
@@ -1349,6 +1558,15 @@ def test_sentinel_hub_sar_backscatter_batch_process(auth_connection, tmp_path):
     output_tiff = result_asset_paths[0]
     assert_geotiff_basics(output_tiff, expected_band_count=4)  # VV, VH, mask and local_incidence_angle
 
+    # TODO: verify this test for projection metadata works.
```
Comment (soxofaan) [inline tests/test_integration.py:1561]: Will this TODO be resolved in this PR? or is that for after merging?

File: tests/test_integration.py [hunk 5]
```diff
@@ -1349,6 +1558,15 @@ def test_sentinel_hub_sar_backscatter_batch_process(auth_connection, tmp_path):
...
+    #   Have not yet been able to verify this part of the test works due to
+    #   jobs timing out and earlier parts of the test failing.
+    #   Therefore, being cautious to avoid having integration tests fail when
```
Comment (soxofaan) [inline tests/test_integration.py:1564]: FYI: you could also do error/warning logging about issues to keep test passing, but still be able to inspect unexpected state of new features

File: tests/test_integration.py [hunk 6]
```diff
@@ -1805,6 +2026,15 @@ def test_load_collection_references_correct_batch_process_id(auth_connection, tm
         for band1, band2 in itertools.combinations([sigma0_vv, sigma0_vh, gamma0_vv, gamma0_vh], 2):
             assert not np.array_equal(band1, band2)
 
+    # TODO: verify this test for projection metadata works, then uncomment it.
```
Comment (soxofaan) [inline tests/test_integration.py:2029]: Same question as above

File: tests/test_integration.py (hunk 5 above)
Comment (soxofaan) [reply tests/test_integration.py:1564]: for example with a wrapper around the experimental bit:
```python
try:
     experimental()
except Exception as e:
     _log.warning("This failed {e!r}", exc_info=True")
```

File: tests/test_integration.py [hunk 7]
```diff
@@ -336,6 +510,19 @@ def elapsed():
     raise RuntimeError("reached max poll time: {e} > {m}".format(e=elapsed(), m=max_poll_time))
 
 
+def assert_job_status(job: BatchJob, expected_status):
+    # If the next assert is going to fail, then first show the logs of the job.
+    if job.status != expected_status:
+        print(
+            "job {j}: did not end with status {expected_status}, but with {job.status=}"
```
[the remark this answers is not in the document]
Comment (JohanKJSchreurs) [reply tests/test_integration.py]: Should be `{job}` indeed. Now corrected.

File: tests/test_integration.py (hunk 3 above)
Comment (JohanKJSchreurs) [reply tests/test_integration.py]: Done.

File: tests/test_integration.py (hunk 6 above)
Comment (JohanKJSchreurs) [reply tests/test_integration.py]: Now logging it as warning for the part that is still experimental.

File: tests/test_integration.py (hunk 4 above)
Comment (JohanKJSchreurs) [reply tests/test_integration.py]: Now logging it as warning for the part that is still experimental.

**Pull request merged**